### ISSUE TITLE
0.5.0: Change of variable type for "settings" to "map(any)" causes failure

### ISSUE THREAD

**Issue by cnocula-peg** [DNXLabs/terraform-aws-eks-lb-controller#17]
### Description. What happened?

The addition of the type `map(any)` for the variable `settings` caused a break in our deployment pipelines due to:
```
Error: Invalid value for module argument
│ 
│   on modules/eks-aws-lbc/main.tf line 10, in module "load_balancer_controller":
│   10:   settings = {
│   11:     "image" : {
│   12:         "repository" : "602401143452.dkr.ecr.eu-central-1.amazonaws.com/amazon/aws-load-balancer-controller"
│   13:     },
│   14:     "vpcId" : var.vpc_id
│   15:   }
│ 
│ The given value is not suitable for child module variable "settings"
│ defined at
│ .terraform/modules/eks-aws-lbc.load_balancer_controller/_variables.tf:76,1-20:
│ all map elements must have the same type.
```

According to https://github.com/hashicorp/terraform/issues/21384#issuecomment-495431269, `map(any)` is not really a "map of anything", but a "map of anything of the same type".

### Steps to reproduce

1. Define your `settings` map with mixed types, e.g.
```
module "load_balancer_controller" {
  source = "git::https://github.com/DNXLabs/terraform-aws-eks-lb-controller.git"

  cluster_identity_oidc_issuer     = var.cluster_oidc_issuer_url
  cluster_identity_oidc_issuer_arn = var.oidc_provider_arn
  cluster_name                          = var.cluster_id

  settings = {
    "image" : {
        "repository" : "602401143452.dkr.ecr.eu-central-1.amazonaws.com/amazon/aws-load-balancer-controller"
    },
    "vpcId" : var.vpc_id
  }
}
```
In this case, one key of type string, one key of type map. 

2.  Execute `terraform plan`

### Expected behavior

Execution passes without any modifications, as the target Helm chart expects/allows nested maps and also has a mixture of strings, arrays, integers etc.

### Actual behavior

Failure due to:
```
Error: Invalid value for module argument
│ 
│   on modules/eks-aws-lbc/main.tf line 10, in module "load_balancer_controller":
│   10:   settings = {
│   11:     "image" : {
│   12:         "repository" : "602401143452.dkr.ecr.eu-central-1.amazonaws.com/amazon/aws-load-balancer-controller"
│   13:     },
│   14:     "vpcId" : var.vpc_id
│   15:   }
│ 
│ The given value is not suitable for child module variable "settings"
│ defined at
│ .terraform/modules/eks-aws-lbc.load_balancer_controller/_variables.tf:76,1-20:
│ all map elements must have the same type.
```

### Occurrence

Frequently.

For the time being, I had to pin the version of `terraform-aws-eks-lb-controller` to 0.4.1 to unblock the pipelines.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caiovfernandes** [comment]
Thanks for reporting this bug to us @cnocula-peg.
I will fix this issue.

**Comment by caiovfernandes** [comment]
Hi @cnocula-peg , we just fix this bug on the release [0.5.1](https://github.com/DNXLabs/terraform-aws-eks-lb-controller/releases/tag/0.5.1).
